pkg/registry/core/pod/strategy.go

```diff
@@ -836,7 +839,10 @@ func validateContainer(container string, pod *api.Pod) (string, error) {
 		}
 	} else {
 		if !podHasContainerWithName(pod, container) {
-			return "", errors.NewBadRequest(fmt.Sprintf("container %s is not valid for pod %s", container, pod.Name))
+			coreV1Pod := &apiv1.Pod{}
+			err := apiscorev1.Convert_core_Pod_To_v1_Pod(pod, coreV1Pod, nil)
+			klog.ErrorS(err, "Pod failed to convert to v1", "pod", klog.KObj(pod))
+			return "", errors.NewBadRequest(fmt.Sprintf("container %s is not valid for pod %s out of: %s", container, pod.Name, podcmd.AllContainerNames(coreV1Pod)))
```

Comment:
I'm afraid that if `err := apiscorev1.Convert_core_Pod_To_v1_Pod(pod, coreV1Pod, nil)` returns error, then we can have issues during `return "", errors.NewBadRequest(fmt.Sprintf("container %s is not valid for pod %s out of: %s", container, pod.Name, podcmd.AllContainerNames(coreV1Pod)))` as `coreV1Pod` will be empty.  Maybe we need a better error handling

Reply:
@ArangoGutierrez Updated with error handling, PTAL thanks.